### PR TITLE
Add domain funding to colony home

### LINKED ISSUE #1783: Add domain funding to colony home
### Story

As a user with funding role in a domain (but not admin in the colony), I want to be able to use the "move funds" dialog, so that I can control the funding for domains in which I have the funding role.

### Description

We've already created a "move funds" dialog in #1780, now we want to add a button to open it from the colony home page, as well as a section displaying the currently allocated funds for the selected domain.

The "Fund" button to open the dialog should be shown if the current user has the funding role in any domain of the colony, otherwise it's hidden. The current domain funds should show the amount for any tokens which have been selected/enabled in the colony admin tokens screen.

If "All" domains is selected, hide the whole available funds section and "Fund" button.

🌮 If this functionality was done in #1780 (or you fancy adding it now!), then the dialog should open with the currently selected domain already selected as the "to" domain. If the current user doesn't have permission, then default back to nothing selected.

### Design

[Figma link](https://www.figma.com/file/L6scZfliLKnDt95nwan4jwzg/DLP-DLF?node-id=998%3A28357)

image

image

### This issue is complete when...

- [ ] "Available Funds" section added to colony home sidebar
- [ ] "Fund" button shown as part of that section, but only if the current user has funding role in at least one domain of the colony - this should open the dialog

### Suggestions

* ???

### Subtasks

- [ ] ???

### PR DESCRIPTION
**Pull request by ceolson01**
## Description

This PR adds a domain funding (both balances and "Fund" modal button) widget to the colony home.

**New stuff** ✨

* `ColonyFunding` component

**Changes** 🏗

* Use `ColonyFunding` component in `ColonyHome`

## TODO

- [x] Wire up to the transfer funds between pots dialog once #1858 is complete

Closes #1783

### CODE REVIEW

**Comment by rdig** [comment]
So I think this should be based on Karolin's #1823 since funding goes on the refreshed Colony Dashboard.

But most likely that will be merged before this, so I guess it's all good :)

**Comment by ceolson01** [comment]
Yes, good call @rdig. I'll add that as a blocker 👍

File: src/modules/dashboard/components/ColonyHome/ColonyHome.tsx [hunk 1]
```diff
@@ -125,7 +113,7 @@ const ColonyHome = ({
   intl: { formatMessage },
 }: Props) => {
   const [filterOption, setFilterOption] = useState(TasksFilterOptions.ALL_OPEN);
-  const [filteredDomainId, setFilteredDomainId] = useState(0);
+  const [filteredDomainId, setFilteredDomainId] = useState(1);
```
Comment (sprusr) [inline src/modules/dashboard/components/ColonyHome/ColonyHome.tsx]: Should it be all or root selected by default? Was there a product decision?

Comment (rdig) [reply src/modules/dashboard/components/ColonyHome/ColonyHome.tsx]: But isn't `root` being selected show all tasks in all it's child domains? Meaning all tasks are displayed regardless ?

Comment (sprusr) [reply src/modules/dashboard/components/ColonyHome/ColonyHome.tsx]: So actually selecting domain `1` will show only tasks from the root domain, and not children. You have to select `0` to show all tasks.

File: src/modules/dashboard/components/ColonyHome/ColonyMeta/ColonyMeta.tsx [hunk 2]
```diff
@@ -146,9 +142,9 @@ const ColonyMeta = ({
           <li>
             <Button
               className={getActiveDomainFilterClass(0, filteredDomainId)}
-              onClick={() => setFilteredDomainId()}
+              onClick={() => setFilteredDomainId(1)}
             >
-              <FormattedMessage {...MSG.allDomains} />
+              <FormattedMessage {...{ id: 'domain.root' }} />
```
[the remark this answers is not in the document]
Comment (ceolson01) [reply src/modules/dashboard/components/ColonyHome/ColonyMeta/ColonyMeta.tsx]: 🤦‍♂

No clue why I did that, nice catch!

**Comment by ceolson01** [comment]
@sprusr might be good to give this another quick look - some changes after rebase, including using `0` for `root` domain.

This `0` & `1` `domainId` business is getting tedious 😅

File: src/modules/dashboard/components/ColonyHome/ColonyFunding/ColonyFunding.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    () =>
+      openDialog('TokensMoveDialog', {
+        colonyAddress,
+        toDomain: currentDomainId,
+      }),
+    [openDialog, colonyAddress, currentDomainId],
+  );
+
+  // hide for root domain
+  return currentDomainId === 0 ? null : (
```
Comment (sprusr) [inline src/modules/dashboard/components/ColonyHome/ColonyFunding/ColonyFunding.tsx]: Hm yeah this is actually kind of confusing behaviour now that I think about it more, but it's what was agreed with product at least for now. I think the whole selecting of domain + hierarchy + "all domains" behaviour needs another look from dev + product point of view.

**Pull request merged**